### PR TITLE
Game and author classes

### PR DESCRIPTION
**Pull request by Olamarx**
- Create a Game class in a separate .rb file.
- Create an Author class with an association to the Item class (in a separate .rb file).
- All Game class properties visible in the diagram are defined and set up in the constructor method.
- All Author class properties visible in the diagram are defined and set up in the constructor method.
- Methods implemented:
    - add_item method in the Author class
        - Takes an instance of the Item class as an input
        - Add the input item to the collection of items
        - Add self as a property of the item object (by using the correct setter from the item object)
    - can_be_archived?() in the Game class
        - Overrides the method from the parent class
        - Returns true if parent's method returns true AND if last_played_at is older than 2 years
        - otherwise, it should return false

### CODE REVIEW

File: lib/author.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Author
+  attr_accessor :first_name, :last_name
+
+  def initialize(first_name = 'Undefined', last_name = 'Undefined')
+    @last_name = last_name
+    @first_name = first_name
+    @items = items
```
Comment (darikmohammed) [inline lib/author.rb:7]: [Required] @item should be an array `@items = items` the variable items is not in the initialized variable. Kindly make `@items = []`

File: lib/game.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+require_relative 'item'
+
+class Game < Item
+  attr_accessor :publisher, :last_played_at
+
+  def initialize(publish_date = 'Undefined', publisher = 'Undefined', last_played_at = 'Undefined', archived: false)
```
Comment (darikmohammed) [inline lib/game.rb:6]: [Required] Game class initializer is missing the multiplayer variable which is public. Kindly please add a multiplayer variable for your initializer.
[Required] last_played_at is a Date data type but it is initialized as String. Kindly fix this. You can use `last_played_at = Time.now.strftime('%d/%m/%Y')`

File: lib/game.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    super(publish_date, archived: archived)
+    @publisher = publisher
+    @last_played_at = last_played_at
+  end
+
+  private
+
+  def can_be_archieved?
+    super && @last_played_at > 2
+  end
```
Comment (darikmohammed) [inline lib/game.rb:16]: [Required] Kindly check the date of the last_played_at with a date conversation not as a regular integer.
- Instead you can use `Date.today.year - Date.parse(@last_played_at).year > 2` when comparing the last played year

**Review by darikmohammed — CHANGES_REQUESTED** [review]
Hi @Olamarx,

Good job so far!

To Highlight! 🟢 
✔️ Good job with clean code
✔️ Great pull request description

There are some issues that you still need to work on to go to the next project but you are almost there! 

image

## Required changes ♻️ 
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

File: lib/game.rb (hunk 3 above)
Comment (Olamarx) [reply lib/game.rb:16]: Oh, I get that. Thank you @darikmohammed.

File: lib/author.rb (hunk 1 above)
Comment (Olamarx) [reply lib/author.rb:7]: @darikmohammed thanks for the review, I will fix it asap.

File: lib/game.rb (hunk 2 above)
Comment (Olamarx) [reply lib/game.rb:6]: @darikmohammed can you please be clear on the first required? I did not see a provided variable name in the requirement which is what confuses me. Kindly be explicit.

For the second, I will fix it up, thank you.

Comment (darikmohammed) [reply lib/game.rb:6]: Screen Shot 2022-09-06 at 11 44 33 PM
@Olamarx , 
In the Game.rb class you should have the variable multiplayer but it is not declared in your case

Comment (Olamarx) [reply lib/game.rb:6]: Oh!!! Now I see, that I was following the requirement on the issue page. The last_play_at was pointed there but not multiplayer.

I will get that done now.

File: lib/game.rb (hunk 3 above)
Comment (Olamarx) [reply lib/game.rb:16]: Done.

File: lib/game.rb (hunk 2 above)
Comment (Olamarx) [reply lib/game.rb:6]: Done

**Review by dqarias — APPROVED** [review]
## Hi @Olamarx  👋,
Good job so far!, This milestone is complete! 💪

There is nothing else to say other than... it's time to merge it :shipit:

## To Highlight 🎉
No linter errors. ✔️
GitHub flow is followed. ✔️
Good job doing the changes required✔️

## Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR.

Cheers and Happy coding!👏👏👏

**Review by darikmohammed — APPROVED** [review]
Hi @Olamarx , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Clean/Organized code

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Comment by Olamarx** [comment]
Thank you guys for the review.

**Pull request merged**